main.tf

```diff
@@ -1,5 +1,5 @@
 resource "aws_key_pair" "key" {
-  count      = var.ssh_key_pair == "" ? 0 : 1
+  count      = var.ssh_public_key == "" ? 0 : 1
```

Comment:
`var.ssh_key_pair` was introduced to re-use existing keys. If set no key will be generated by the module. Your change looks like doing the same but for some reason the default example fails no.

```
Error: Invalid count argument

  on ../../main.tf line 2, in resource "aws_key_pair" "key":
   2:   count      = var.ssh_public_key == "" ? 0 : 1

The "count" value depends on resource attributes that cannot be determined
until apply, so Terraform cannot predict how many instances will be created.
To work around this, use the -target argument to first apply only the
resources that the count depends on.
```

Reply:
I solved the problem with key on in a slight other way.

Reply:
@npalm so what we should leave in this PR and what should be removed to resolve the conflict?